src/openzaak/tests/test_enabled_middleware.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2025 Dimpact
+from unittest.mock import patch
+
+from django.http import HttpResponseNotFound
+from django.test import RequestFactory, SimpleTestCase
+
+from openzaak.utils.middleware import EnabledMiddleware
+
+
+class EnabledMiddlewareDatabaseQueryTest(SimpleTestCase):
```

Comment:
Generally I'd prefer to mock as little as possible, so instead of using mocks here, let's use the database
```suggestion
class EnabledMiddlewareDatabaseQueryTest(TestCase):
```

You can then do something like:

```python
    def test_no_db_query_if_no_component_type(self):
        request = self.factory.get("/")

        with self.assertNumQueries(0):
            result = self.middleware.process_view(request, None, None, None)

        self.assertIsNone(result)
```

And for the 404 test you'd have to ensure the `InternalService` you're testing is disabled. You could also add a third test to verify that `InternalService.enabled=True` works as expected, that way all cases are covered.